security/security-design/shared-assets/kms-backup/files/kms_key_backup.py

```diff
@@ -0,0 +1,489 @@
+def load_config(config_file_name):
+    """Load config file and returns dict of configuration"""
+    try:
+        config_file = open(config_file_name)
+    except Exception as ioError:
+        print("Load of configuration file: " + args.configfile + " failed")
+        exit(1)
+    try:
+        config = json.load(config_file)
+        config_file.close()
+    except Exception as ConfigError:
+        print("JSON Parse of configuration file: " + config_file_name + " failed")
+        exit(1)
+    return config
+
+#########################################################################
+# parse_cmd_line
+# Parses the command line with argsparser and returns args Namespace
+#########################################################################
+def parse_cmd_line():
+    """ parse_cmd_line """
+    global config_parameters
+    args_parser = argparse.ArgumentParser(
+        description='OCI KMS Software key copy program'
+    )
+    #
+    # Loop throug and set all commandline parameters
+    #
+    for arg in config_parameters:
+        if not arg[2]:
+            args_parser.add_argument('--'+arg[0],default=arg[1],type=str,help=arg[3])
+        else:
+            args_parser.add_argument('--'+arg[0],default=arg[1],action='store_true',help=arg[3])
+    # Parse and return parsed Namespace
+    args = args_parser.parse_args()
+    return(args)
+
+#########################################################################
+# overwrite_config
+# Overwrite configfile settings from command line
+# Any commandline settings will overwrite any settings inherited from the 
+# config file, if loaded
+#########################################################################
+def overwrite_conf(args,config):
+    """ overwrite_conf """
+    dargs=vars(args)
+    for arg in dargs.keys():
+        if dargs[arg] is not None:
+            config[arg]=dargs[arg]
+    if 'exportonly' not in dargs or dargs['exportonly'] is None or not dargs['exportonly']:
+        config['exportonly']=False
+    else:
+        config['exportonly']=True
+
+#########################################################################
+# verify_config
+# Verifies that all mandatory parameters are set
+# Take the two modi emxortkey or copykey into account
+#########################################################################
+def verify_config(config):
+    """ verify_config, verifies that all mandatory configuration parameters is set """
+    #
+    # Mandatory source parameters
+    # source_keyname" or"source_key_ocid"checked for explisitt
+    #
+    mandatory_source_parameters = [ 'ociconfig','source_ociprofile','source_region','source_compartment','source_vault']
+    #
+    # Mandatory target parameters
+    #
+    mandatory_target_parameters=['target_region','target_compartment','target_vault','target_ociprofile', 'target_keyname']
+    #
+    # Mandatory target parameters
+    #
+    mandatory_other_parameters=['wrapping_algorithm']
+    #
+    complete=0

```

Comment:
This can be handled by the argsparse library when providing the 'required=True' flag.

Reply:
Slightly different semantic, you may have the parameter either in the config file or as a CMD line parameter